lib/frontend.ml

```diff
@@ -163,7 +164,8 @@ module Make(C: S.CONFIGURATION with type 'a io = 'a Lwt.t) = struct
       let pages = Io_page.pages num in
       List.iter
         (fun (gref, page) ->
-           let id = gref mod (1 lsl 16) in
+           let id = nf.rx_id in
+           nf.rx_id <- (succ nf.rx_id) mod (1 lsl 16) ;
```

Comment:
while I agree that this might happen (but IMHO it is very unlikely), what would your proposal for the id be? invoking `Random.int`? keeping a list/array of free (or used) `ids` seems overengineered to me.